### PR TITLE
Feat(install tabs): introduce

### PR DESCRIPTION
**Pull request by AugustinMauroy**
## Description

Adding `installTabs`

In nodejs.dev the style of installTabs is not the same because style is "sparated" on par in `installTabs` others part is on `pages/index.tsx` so I have merge the style to have this component better.
But I don't have take leafs in this component

## Related Issues

Related issues: [Migrate DownloadComponents and Hero Component from nodejs.dev #5303](https://github.com/nodejs/nodejs.org/issues/5303)

### Check List

- [X] I have read the [Contributing Guidelines](https://github.com/nodejs/nodejs.org/blob/main/CONTRIBUTING.md) and made commit messages that follow the guideline.
- [X] I have run `npx turbo lint` to ensure the code follows the style guide. And run `npx turbo lint:fix` to fix the style errors if necessary.
- [X] I have run `npx turbo format` to ensure the code follows the style guide.
- [X] I have run `npx turbo test` to check if all tests are passing, and/or `npx turbo test:snapshot` to update snapshots if I created and/or updated React Components.
- [x] I've covered new added functionality with unit tests if necessary.

### CODE REVIEW

**Comment by AugustinMauroy** [comment]
I'll fix conflict

**Review by mikeesto — COMMENTED** [review]
Hi @AugustinMauroy . I'm not sure exactly where you are at with this. FYI, I've run the PR checks and there's a small issue with the tests.

This also looks like it is going to be impacted by https://github.com/nodejs/nodejs.org/pull/5425 ? (assuming it lands)

**Comment by AugustinMauroy** [comment]
Hey ! I will take a look to test.
But for impact of the other PR. Normally this has no impact, as the `.scss` style overrides `prism`.

File: .github/ISSUE_TEMPLATE/01-bug-report.yml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 name: Report a Technical/Visual Issue on the Node.js Website
-description: "Is something not working as expected? Did you encounter a glitch or a bug with the Website?"
+description: 'Is something not working as expected? Did you encounter a glitch or a bug with the Website?'
```
Comment (rodion-arr) [inline .github/ISSUE_TEMPLATE/01-bug-report.yml:2]: While this PR is on tabs components, why do we have changes in `.github` folders?

File: components/Home/InstallTabs/LinuxPanel/index.tsx [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        <span className={styles.installTextNoSelect}>$</span>
+        <span className={styles.installTextCommand}>curl -o- </span>
+        {nvmInstallScriptUrl}
+        <span className={styles.installTextCommand}> | bash</span>
+      </ShellBox>
+      <ShellBox textToCopy="nvm install --lts">
+        <span className={styles.installTextNoSelect}>$</span>
+        <span className={styles.installTextCommand}>nvm</span> install --lts
+      </ShellBox>
+      {/* @TODO: Verify this link when we implement content */}
```
Comment (rodion-arr) [inline components/Home/InstallTabs/LinuxPanel/index.tsx:28]: Do we have a process of handling todos in the code? I'm afraid this todo comment may stay here forever

File: components/Home/InstallTabs/index.tsx
Comment (rodion-arr) [inline components/Home/InstallTabs/index.tsx:1]: There is a strange "jumping" issue on long code line hover, I don't see this on old `nodejs.dev` site, could you please check it


https://github.com/nodejs/nodejs.org/assets/5843270/e9065de3-101c-46b1-bfe7-9bd67595bbef

File: .github/ISSUE_TEMPLATE/01-bug-report.yml (hunk 1 above)
Comment (AugustinMauroy) [reply .github/ISSUE_TEMPLATE/01-bug-report.yml:2]: That form an `npm run format`

File: components/Home/InstallTabs/LinuxPanel/index.tsx (hunk 2 above)
Comment (AugustinMauroy) [reply components/Home/InstallTabs/LinuxPanel/index.tsx:28]: I don't know but yeah todo is dangerous

File: components/Home/InstallTabs/index.tsx
Comment (AugustinMauroy) [reply components/Home/InstallTabs/index.tsx:1]: I will check that but it's will change whit [this pr](https://github.com/nodejs/nodejs.org/pull/5425)

File: components/Home/InstallTabs/LinuxPanel/index.tsx [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+type Props = {
+  nvmVersion: string;
+};
+
+export const PureLinuxPanel: FC<Props> = ({ nvmVersion }) => {
+  const nvmInstallScriptUrl = `https://raw.githubusercontent.com/nvm-sh/nvm/${nvmVersion}/install.sh`;
+
+  return (
+    <div>
```
Comment (shanpriyan) [inline components/Home/InstallTabs/LinuxPanel/index.tsx]: Do we need this <div> here for layout , can we make it a `<></>` Fragment if not being used in layout?

File: components/Home/InstallTabs/LinuxPanel/index.tsx [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+      <ShellBox textToCopy="nvm install --lts">
+        <span className={styles.installTextNoSelect}>$</span>
+        <span className={styles.installTextCommand}>nvm</span> install --lts
+      </ShellBox>
+      {/* @TODO: Verify this link when we implement content */}
+      <LocalizedLink
+        className={styles.installDocsButton}
+        href="/download/package-manager/#nvm"
+      >
+        <FormattedMessage id="components.home.installTabs.readDocs" />
```
Comment (shanpriyan) [inline components/Home/InstallTabs/LinuxPanel/index.tsx:33]: Shouldn't we update the key as per guidelines? 🤔 
```suggestion
        <FormattedMessage id="components.home.installTabs.linuxPanel.readDocs" />
```

File: components/Home/InstallTabs/LinuxPanel/index.tsx [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+      </LocalizedLink>
+    </div>
+  );
+};
+
+const LinuxPanel: FC = () => {
+  const [nvmVersion, setNvmVersion] = useState('');
+
+  useEffect(() => {
+    getNvmLatestVersion().then(version => setNvmVersion(version));
```
Comment (shanpriyan) [inline components/Home/InstallTabs/LinuxPanel/index.tsx]: nit
```suggestion
    getNvmLatestVersion().then(setNvmVersion);
```

File: components/Home/InstallTabs/LinuxPanel/index.tsx (hunk 4 above)
Comment (AugustinMauroy) [reply components/Home/InstallTabs/LinuxPanel/index.tsx:33]: we will not have 3 key ?

File: components/Home/InstallTabs/LinuxPanel/index.tsx (hunk 2 above)
Comment (AugustinMauroy) [reply components/Home/InstallTabs/LinuxPanel/index.tsx:28]: @ovflowd Have you an idea to secure all todo ?

File: components/Home/InstallTabs/LinuxPanel/index.tsx (hunk 3 above)
Comment (AugustinMauroy) [reply components/Home/InstallTabs/LinuxPanel/index.tsx]: Fragment is better yes8

File: components/Home/InstallTabs/LinuxPanel/index.tsx (hunk 4 above)
Comment (AugustinMauroy) [reply components/Home/InstallTabs/LinuxPanel/index.tsx:33]: I thinks it's bad idea. But thanks for the reflexion

File: components/Home/InstallTabs/index.tsx [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+      addPanel(<WindowsPanel />);
+      break;
+    default:
+      addPanel(<WindowsPanel />);
+      addPanel(<MacOSPanel />);
+      addPanel(<LinuxPanel />);
+      break;
+  }
+
+  return <>{panels}</>;
```
Comment (shanpriyan) [inline components/Home/InstallTabs/index.tsx]: little Refactor
```suggestion
  const createPanel = (...components: ReactNode[]) => {
  return components.map((comp) => (
    <TabPanel {...tabPanelStyles}>{comp}</TabPanel>
  ));
};

switch (userOS) {
  case "MAC":
    return createPanel(<MacOSPanel />, <WindowsPanel />, <LinuxPanel />);
  case "LINUX":
    return createPanel(<LinuxPanel />, <MacOSPanel />, <WindowsPanel />);
  default:
    return createPanel(<WindowsPanel />, <MacOSPanel />, <LinuxPanel />);
}

```

File: components/Home/InstallTabs/index.tsx
Comment (ovflowd) [reply components/Home/InstallTabs/index.tsx:1]: I would also argue that these two snippets (two different lines) should be only one Codebox, and not two 👀

File: components/Home/InstallTabs/LinuxPanel/index.tsx (hunk 2 above)
Comment (ovflowd) [reply components/Home/InstallTabs/LinuxPanel/index.tsx:28]: What is this TODO supposed to be?

**Comment by ovflowd** [comment]
Also, @AugustinMauroy too many unrelated changes within this PR, I'd ask if possible to recreate the branch 🙈

**Comment by AugustinMauroy** [comment]
@ovflowd why not

**Pull request closed without merging**